### PR TITLE
Allow empty string to be passed as 'uuid' param

### PR DESCRIPTION
**Pull request by spellr**
This is for APIs that accept UUID, but want to be able to support
passing empty string for deleting the uuid attribute.

None is obviously not good enough, since there's no way to distinguish
between 'none has been given' or 'nothing has been given'.

### CODE REVIEW

File: hammock/converters.py [hunk 1]
```diff
@@ -68,6 +68,9 @@ def to_unicode(value):
 
 
 def to_uuid(value):
+    # Empty string can be passed as a UUID
```
Comment (lior-stratoscale) [inline hammock/converters.py]: there are many uses of this parameter value in the system. are you sure this is better than creating a custom type for your use?

Comment (spellr) [reply hammock/converters.py]: Custom type `uuid[None]`? Don't like it.. What's wrong with supporting empty string on `uuid`?

**Comment by ravidbro** [comment]
@spellr can you give an example of an API that will use that? How will the URL look like?

**Comment by spellr** [comment]
@ravidbro This is an old change and I don't remember the exact place did it for.
It's meant for body params, not for url params.
It's useful distinguishing between None - which may be the default behavior (i.e. attach the default network), and empty string (i.e. attach no network)

**Comment by spellr** [comment]
@ravidbro In the query string, it can be:
GET /vms?project_id=&network_id=

**Comment by ravidbro** [comment]
Thanks

**Pull request merged**